build.zig

```diff
@@ -555,6 +555,11 @@ fn link_tracer_backend(
                 exe.linkSystemLibrary("dbghelp");
                 exe.linkSystemLibrary("ws2_32");
             }
+
+            // We might need to clone Tracy, if it doesn't exist
+            const git_clone = exe.builder.addSystemCommand(&.{ "sh", "-c", "test -d tools/tracy ||" ++
```

Comment:
I'd rather we avoid `sh` here -- it doesn't seem to be _fundamentally_ required, as we are full-fledged Zig process at this point and can do whatever. It might be shorter to write in the short-term, but long term it would be faster to start building a library for scripting in Zig perhaps.

OTOH, we _already_ use "sh" in the build.zig, so this doesn't introduce a new dependency. I guess I'll look into expunging sh from build.zig after this is merged, along the lines of 

https://github.com/matklad/xshell

Reply:
Yeah, it was my laziness coming through here :laughing:. All I'm using it for is to check if a dir exists, which is simple enough in pure Zig. I'll refactor this and the other instance to do that.

Reply:
I have some wrappers for running commands cross platform in shells in src/clients/shutil.zig already as well.

Reply:
Thanks for the heads up @eatonphil - I've modified `file_or_directory_exists` to no longer need an allocator, and using it now

Reply:
A draft for further reducing sh dependencies https://github.com/tigerbeetledb/tigerbeetle/pull/950.